### PR TITLE
HIP on Windows initial support

### PR DESCRIPTION
**Pull request by stanleytsang-amd**
Initial HIP on Windows support for hipCUB. Currently on my test machine hipCUB builds, installs, and unit tests pass.

How to build with HIP on Windows:

Get system dependencies with rdeps.py: python rdeps.py
Install using new rmake.py script: python rmake.py
To run tests: python rtest.py -t <insert test set name here, see rtest.xml for test set names>

I have not tested rmake.py with linux yet, I suspect it still needs some work, but the currently install.sh script works just as it did before for linux. Long-term we can decide what we want to do with install.sh/rmake.py.

### CODE REVIEW

File: test/hipcub/test_hipcub_device_segmented_radix_sort.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
+#include <numeric>
```
Comment (neon60) [inline test/hipcub/test_hipcub_device_segmented_radix_sort.cpp:23]: I think we can remove this line.

File: test/hipcub/test_utils.hpp [hunk 2]
```diff
@@ -65,6 +65,59 @@ inline auto get_random_data(size_t size, T min, T max, int seed_value)
     return data;
 }
 
+#if defined(_WIN32) && defined(__clang__)
+#if 0
```
Comment (neon60) [inline test/hipcub/test_utils.hpp:69]: Can we add a comment why we need these unused lines?

**Comment by neon60** [comment]
Please update readme and changelog.

File: test/hipcub/test_utils.hpp (hunk 2 above)
Comment (stanleytsang-amd) [reply test/hipcub/test_utils.hpp:69]: Sorry, these were leftover lines.   Removed.

**Comment by stanleytsang-amd** [comment]
> Please update readme and changelog.

Done.

File: README.md [hunk 3]
```diff
@@ -18,6 +18,9 @@ environment, hipCUB uses the rocPRIM library as the backend.  However, on CUDA p
 * For NVIDIA GPUs:
   * CUDA Toolkit
   * CUB library (automatically downloaded and by CMake script)
+* Python 3.6 or higher (HIP on Windows only)
```
Comment (saadrahim) [inline README.md:21]: I the requirement is for the build scripts only. We have no hard python requirements.

**Review by saadrahim — APPROVED** [review]
note the python requirements comment

File: README.md (hunk 3 above)
Comment (stanleytsang-amd) [reply README.md:21]: good point. i'll edit the READMEs to be more explicit that python is for the build scripts.

**Pull request merged**